src/error.rs

```diff
@@ -284,19 +276,330 @@ impl From<InvalidWindowHandle> for CmdError {
+            UnableToCaptureScreen => StatusCode::BAD_REQUEST,
+            UnableToSetCookie => StatusCode::INTERNAL_SERVER_ERROR,
+            UnexpectedAlertOpen => StatusCode::INTERNAL_SERVER_ERROR,
+            UnknownCommand => StatusCode::NOT_FOUND,
+            UnknownError => StatusCode::INTERNAL_SERVER_ERROR,
+            UnknownMethod => StatusCode::METHOD_NOT_ALLOWED,
+            UnknownPath => StatusCode::NOT_FOUND,
+            UnsupportedOperation => StatusCode::INTERNAL_SERVER_ERROR,
+        }
+    }
+}
+
+impl Serialize for ErrorStatus {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        self.error_code().serialize(serializer)
+    }
+}
+
+impl From<webdriver::ErrorStatus> for ErrorStatus {
+    fn from(e: webdriver::ErrorStatus) -> Self {
+        use self::webdriver::ErrorStatus::*;
+        match e {
+            DetachedShadowRoot => ErrorStatus::DetachedShadowRoot,
+            ElementClickIntercepted => ErrorStatus::ElementClickIntercepted,
+            ElementNotInteractable => ErrorStatus::ElementNotInteractable,
+            ElementNotSelectable => ErrorStatus::ElementNotSelectable,
+            InsecureCertificate => ErrorStatus::InsecureCertificate,
+            InvalidArgument => ErrorStatus::InvalidArgument,
+            InvalidCookieDomain => ErrorStatus::InvalidCookieDomain,
+            InvalidCoordinates => ErrorStatus::InvalidCoordinates,
+            InvalidElementState => ErrorStatus::InvalidElementState,
+            InvalidSelector => ErrorStatus::InvalidSelector,
+            InvalidSessionId => ErrorStatus::InvalidSessionId,
+            JavascriptError => ErrorStatus::JavascriptError,
+            MoveTargetOutOfBounds => ErrorStatus::MoveTargetOutOfBounds,
+            NoSuchAlert => ErrorStatus::NoSuchAlert,
+            NoSuchCookie => ErrorStatus::NoSuchCookie,
+            NoSuchElement => ErrorStatus::NoSuchElement,
+            NoSuchFrame => ErrorStatus::NoSuchFrame,
+            NoSuchShadowRoot => ErrorStatus::NoSuchShadowRoot,
+            NoSuchWindow => ErrorStatus::NoSuchWindow,
+            ScriptTimeout => ErrorStatus::ScriptTimeout,
+            SessionNotCreated => ErrorStatus::SessionNotCreated,
+            StaleElementReference => ErrorStatus::StaleElementReference,
+            Timeout => ErrorStatus::Timeout,
+            UnableToCaptureScreen => ErrorStatus::UnableToCaptureScreen,
+            UnableToSetCookie => ErrorStatus::UnableToSetCookie,
+            UnexpectedAlertOpen => ErrorStatus::UnexpectedAlertOpen,
+            UnknownCommand => ErrorStatus::UnknownCommand,
+            UnknownError => ErrorStatus::UnknownError,
+            UnknownMethod => ErrorStatus::UnknownMethod,
+            UnknownPath => ErrorStatus::UnknownPath,
+            UnsupportedOperation => ErrorStatus::UnsupportedOperation,
+        }
+    }
+}
+
+impl From<&str> for ErrorStatus {
```

Comment:
I think `FromStr` (and possibly `TryFrom<&str>`) are the traits to implement instead. `From` should be infallible, and `UnknownError` feels like a cop-out.